crates/dbsp/src/trace/mod.rs

```diff
@@ -130,6 +130,21 @@ impl<T> DBWeight for T where T: DBData + MonoidValue {}
 pub trait DBTimestamp: DBData + Timestamp {}
 impl<T> DBTimestamp for T where T: DBData + Timestamp {}
 
+pub trait FilterFunc<V>: Fn(&V) -> bool {
+    fn fork(&self) -> Filter<V>;
```

Comment:
Is this a factory?
Maybe you should call it "factory", and the method create?

Reply:
It's not a factory, just a closure. `fork` is just `clone`, but `clone` is not object-safe, so I needed an alias.

Reply:
A closure is a way of implementing a factory.
This really looks like an object that gives you a filter.

Reply:
This is a closure you apply to a value to determine if it should be retained. A factory is an object that produces entity of some other type.  This thing only clones itself.